contracts/StrategyCurveEcrv.sol

```diff
@@ -104,38 +108,45 @@ contract StrategyCurveEcrv is BaseStrategy {
         _liquidatedAmount = Math.min(_amountNeeded, want.balanceOf(address(this)));
     }
 
-    // NOTE: Can override `tendTrigger` and `harvestTrigger` if necessary
-
     function prepareMigration(address _newStrategy) internal override {
-        // TODO: Transfer any non-`want` tokens to the new strategy
-        // NOTE: `migrate` will automatically forward all `want` in this strategy to the new one
-        prepareReturn(proxy.balanceOf(gauge));
+        uint256 gaugeTokens = proxy.balanceOf(gauge);
+        if (gaugeTokens > 0) {
+            proxy.withdraw(gauge, address(want), gaugeTokens);
+        }
     }
 
     function protectedTokens() internal view override returns (address[] memory) {
-        address[] memory protected = new address[](1);
+        address[] memory protected = new address[](2);
         protected[0] = gauge;
+        protected[1] = address(crv);
 
         return protected;
     }
 
     //
-    // helper functions
+    // setters
     //
-    function setCRVRouter(bool isUniswap, address[] calldata _wethPath) public onlyGovernance {
+    function setCRVRouter(bool isUniswap, address[] calldata _wethPath) external onlyAuthorized {
```

Comment:
Remove the calldata _wethPath argument, just leaving the feature of switching dexes, path is the same on both uni and sushi. And contract can be migrated as well